### DISCUSSION TITLE
use L2 ，does the distance returned by Milvus not perform the sqrt() operation

### DISCUSSION THREAD

**Discussion by liuyichen2013** [milvus-io/milvus#18987] [Q&A and General discussion]
version:milvus2.1.1
use L2 ，does the distance returned by Milvus not perform the sqrt() operation?
In addition，in image similarity search，which way is better ，L2 or IP？

**Comment by yhmo** [comment]
As we know, the Euclidian distance formula has a square root calculation. For similarity comparison, what we need is only a rank, the square root calculation doesn't affect the rank. So, the search engine only returns the value without square root calculation, to improve performance.

**For normalized vectors**, Euclidean distance (L2) is mathematically equivalent to the inner product (IP). If they return different results, check to see if your vectors are normalized.

**Reply by originyy** [reply]
how to check your vectors are normalized

**Reply by yhmo** [reply]
> how to check your vectors are normalized

Assume a vector is [x1, x2, x3, .... xn]. If the vector is normalized, there must be: x1^2 + x2^2 + x3^2 + ... + xn^2 = 1.0
